geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/test/scala/org/locationtech/geomesa/parquet/CompactionTest.scala

```diff
@@ -93,7 +101,18 @@ class CompactionTest extends Specification with AllExpectations {
       fsStorage.metadata.getPartition(partition).map(_.files) must beSome(haveSize[Seq[StorageFile]](2))
       fsStorage.compact(Some(partition))
       fsStorage.metadata.getPartition(partition).map(_.files) must beSome(haveSize[Seq[StorageFile]](1))
-      SelfClosingIterator(fsStorage.getReader(Query.ALL, Some(partition))).toList must haveSize(2)
+      SelfClosingIterator(fsStorage.getReader(Query.ALL, Some(partition))).toList must haveSize(299)
+
+      // compact to a given file size
+      // verify if file is appropriately sized, it won't be modified
+      val paths = fsStorage.getFilePaths(partition).map(_.path)
+      val size = paths.map(f => fc.getFileStatus(f).getLen).sum
+      fsStorage.compact(Some(partition), Some(size))
+      fsStorage.getFilePaths(partition).map(_.path) mustEqual paths
+      // verify files are split into smaller ones
+      fsStorage.compact(Some(partition), Some(size / 2))
+      fsStorage.metadata.getPartition(partition).map(_.files.length).getOrElse(0) must beGreaterThan(1)
+      SelfClosingIterator(fsStorage.getReader(Query.ALL, Some(partition))).toList must haveSize(299)
```

Comment:
I didn't read closely enough.  What's going on with the feature count (seemingly) going from 300 to 299?

Reply:
there's a feature being deleted up on lines 96-100